tests/convert_valid_utf16be_to_utf8_tests.cpp

```diff
@@ -16,6 +16,12 @@ namespace {
   constexpr int trials = 1000;
 }
 
+TEST(valid_but_out_of_range_input)
+{
+    const char16_t input = 0x2020;
+    char out;
+    ASSERT_EQUAL(0, implementation.convert_valid_utf16le_to_latin1(&input, 1, &out));
```

Comment:
You are providing an input that breaks the assumption of the function.

<img width="696" alt="Screenshot 2024-08-11 at 2 58 57 PM" src="https://github.com/user-attachments/assets/00b9f0f8-e49c-4542-9262-cd3e600d8166">